lib/msf/core/module/note.rb

```diff
@@ -0,0 +1,86 @@
+# -*- coding: binary -*-
+
+module Msf
+
+  ###
+  #
+  # Some modules require an extra annotation to provide extra information useful when searching or referencing.
+  # The NotesContainer holds various Note objects that might pertain to an AKA name, an explanation as to why a module
+  # lacks a CVE (NOCVE), or other data.
+  #
+  ###
+
+  class NoteContainer < Hash
+
+    def initialize(opts = {})
```

Comment:
> The NotesContainer holds various Note objects that might pertain to an AKA name, an explanation as to why a module lacks a CVE (NOCVE), or other data.
How does one store other data in the `NoteContainer`? It looks like it is currently fixed with `AKA` and `NOCVE` types.

Reply:
You would do that by creating a new subclass of the `Note` type. This was originally done as an effort to introduce some typing and have some sort of expectation as to what kind of data we're working with—especially since future note-related descriptors could possibly be complex data models. We don't have any of that right now, so I'm open to changing this.

Reply:
I appreciate the typing ability in this design. Is there any reason we wouldn't want to allow those that don't match a specific type to be created as the base `Module::Note` type?